contracts/stellar-example/src/tests/test.rs

```diff
@@ -252,3 +253,83 @@ fn its_example() {
     let recipient = Address::from_string_bytes(&data);
     assert_eq!(token.balance(&recipient), amount);
 }
+
+#[test]
+fn constructor_succeeds() {
+    let env = Env::default();
+
+    let TestConfig {
+        gateway_client,
+        gas_service_client,
+        its_client,
+        ..
+    } = setup_app(&env);
+
+    let contract_id = env.register(
+        Example,
+        (
+            &gateway_client.address,
+            &gas_service_client.address,
+            &its_client.address,
+        ),
+    );
+    let client = ExampleClient::new(&env, &contract_id);
+
+    assert_eq!(client.gateway(), gateway_client.address);
+    assert_eq!(client.gas_service(), gas_service_client.address);
+    assert_eq!(client.interchain_token_service(), its_client.address);
+}
+
+#[test]
+fn execute_fails_with_not_approved() {
+    let env = Env::default();
+
+    let TestConfig { app, .. } = setup_app(&env);
+
+    let source_chain = String::from_str(&env, "ethereum");
+    let message_id = String::from_str(&env, "test");
+    let source_address = String::from_str(&env, "0x123");
+    let payload = Bytes::from_array(&env, &[1u8; 32]);
+
+    assert_contract_err!(
+        app.try_execute(&source_chain, &message_id, &source_address, &payload),
+        ExampleError::NotApproved
+    );
+}
+
+#[test]
+fn send_token_succeeds() {
```

Comment:
`its_example` should do both send and receive part instead, similar to `gmp_example`. We want to show case a full integration example with ITS, rather than a unit test